test/integration/integration_test.go

```diff
@@ -0,0 +1,310 @@
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Pulling",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Pulling image \"" + testpod.Spec.Containers[0].Image + "\"",
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+		{
+			Summary:  "Successfully pulled image \"" + testpod.Spec.Containers[0].Image + "\"",
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Pulled",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Successfully pulled image \"" + testpod.Spec.Containers[0].Image + "\"",
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+		{
+			Summary:  "Created container " + testpod.Spec.Containers[0].Name,
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Created",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Created container " + testpod.Spec.Containers[0].Name,
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+		{
+			Summary:  "Started container " + testpod.Spec.Containers[0].Name,
+			Category: "kubernetes",
+			Attributes: map[string]interface{}{
+				"event.metadata.name":             testpod.Name + ".",
+				"event.metadata.namespace":        ns.Name,
+				"event.reason":                    "Started",
+				"clusterName":                     "",
+				"event.involvedObject.apiVersion": "",
+				"event.involvedObject.kind":       "Pod",
+				"event.involvedObject.name":       testpod.Name,
+				"event.message":                   "Started container " + testpod.Spec.Containers[0].Name,
+				"event.type":                      "Normal",
+				"verb":                            "ADDED",
+			},
+		},
+	} {
+		if agentMock.Has(&event) {
+			continue
+		}
+
+		e := json.NewEncoder(os.Stderr)
+		t.Log("Expected:")
+		_ = e.Encode(event)
+		t.Log("Have:")
+		_ = e.Encode(agentMock.Events())
+		t.Fatalf("Event was not captured")
+	}
+}
+
+func Test_Sink_receives_common_Pod_deletion_events(t *testing.T) {
```

Comment:
I think those 2 tests can be consolidated, so you only need to create a single pod and test namespace. It will save a lot of code and separate "prepare" step from actual assertions.